lib/mint/http1/request.ex

```diff
@@ -22,7 +22,9 @@ defmodule Mint.HTTP1.Request do
   end
 
   defp encode_headers(headers) do
-    Enum.reduce(headers, "", fn {name, value}, acc ->
+    headers
+    |> Enum.map(fn {name, value} -> {name, IO.iodata_to_binary(value)} end)
+    |> Enum.reduce("", fn {name, value}, acc ->
```

Comment:
We don't need to loop through the list of headers twice. Please change this to:

```suggestion
    Enum.reduce(headers, "", fn {name, value}, acc ->
      value = IO.iodata_to_binary(value)
```